lib/binding/kvdb_interface.c

```diff
@@ -165,7 +167,8 @@ hse_init(const char *const config, const size_t paramc, const char *const *const
 
 out:
     if (err) {
-        hse_platform_fini();
+        if (platform_initialized)
+            hse_platform_fini();
```

Comment:
If we make all the xxx_fini() functions idempotent I think it could really simplify setup/teardown...

Reply:
Meaning they clean up after themselves on failure?

Reply:
Yeah, most fini() functions can tell if their subsystem has been initialized, they just have to ensure they do nothing if they haven't been initialized.  Knowing they can only be called single-threaded makes it pretty easy, I think...
They code that fails in the middle of initialization can just call the big "teardown" function regardless of how far along it get, no need to a bunch of goto's and other ugly code :-)

Reply:
A sound approach.  But the `_init` function should clean up after themselves.

Reply:
If you want to get all mathy, a function f(x) is idempotent if f(f(x)) = f(x) for all x that are elements of the input set.  :-)

Reply:
Yep, platform init should take the same approach, which means all the subsystem it initializes must also do the same..
It's turtles all the way down!
Ah, Steve, there's the rub:  The fini() functions are technically not functions, they're procedures!!!